fluent/client/ws_client.go

```diff
@@ -142,25 +179,21 @@ func (c *WSClient) Reconnect() (err error) {
 		err = c.Connect()
 	}
 
+	c.setErr(err)
+
 	return
 }
 
 // SendMessage sends a single msgp.Encodable across the wire.
 func (c *WSClient) SendMessage(e msgp.Encodable) error {
-	if c.Session == nil {
-		return errors.New("No active session")
+	if err := c.getErr(); err != nil {
```

Comment:
These next lines are the meat of the fix. If the connection has errored out, then it will get returned here. If the connection was otherwise closed, it will error below.

Reply:
shouldn't be a connection error returned while we call `c.Session.Connection.Listen()`

Reply:
@ericmillin 

Just a small observation:

We can skip calling `c.Session.Connection.Listen()` if read handler is not set `opts.ReadHandler == nil` it just specifies we are not interested in reading messages from the server. So listening to the server is just overhead.

Reply:
@gufranmirza Without Listen, it won't receive Close messages

Reply:
Got it, thanks